src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/usersettings/changesignature/ChangeSignatureService.java

```diff
@@ -0,0 +1,56 @@
+package de.dhbw.karlsruhe.students.mailflow.core.application.usersettings.changesignature;
+
+import de.dhbw.karlsruhe.students.mailflow.core.application.auth.AuthSessionUseCase;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.value_objects.Address;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.user.UserSettings;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.user.UserSettingsRepository;
+import de.dhbw.karlsruhe.students.mailflow.external.infrastructure.authorization.SaveSettingsException;
+
+/**
+ * @author seiferla
+ */
+public class ChangeSignatureService implements ChangeSignatureUseCase {
+
+  private final UserSettingsRepository userSettingsRepository;
+  private final AuthSessionUseCase authSession;
+
+  public ChangeSignatureService(
+      UserSettingsRepository userSettingsRepository, AuthSessionUseCase authSession) {
+    this.userSettingsRepository = userSettingsRepository;
+    this.authSession = authSession;
+  }
+
+  @Override
+  public void updateSignature(String newSignature)
+      throws LoadSettingsException, SaveSettingsException {
+    UserSettings userSettings =
+        new SettingsBuilder()
+            .withSignature(newSignature)
+            .withAddress(authSession.getSessionUserAddress())
+            .build();
+    userSettingsRepository.updateUserSettings(userSettings);
+  }
```

Comment:
```suggestion
    checkIfLoggedIn();
    UserSettings currentUserSettings =
        userSettingsRepository.getSettings(authSession.getSessionUserAddress());
    UserSettings updatedUserSettings =
        new SettingsBuilder(currentUserSettings).withSignature(newSignature).build();
    userSettingsRepository.updateUserSettings(updatedUserSettings);
  }
```

only update the signature ant not the whole settings here

Reply:
When you've updated this, you can change the resetSignature() method to 

```java
 public void resetSignature() throws LoadSettingsException, SaveSettingsException {
    updateSignature("");
  }
```